src/indra_cogex/representation.py

```diff
@@ -497,39 +504,16 @@ def indra_stmts_from_relations(rels: Iterable[Relation],
     """
     stmts_json = [load_statement_json(rel.data["stmt_json"]) for rel in rels]
     stmts = stmts_from_json(stmts_json)
-    # Beliefs are not set correctly in the JSON so we fix them here
-    beliefs = [rel.data["belief"] for rel in rels]
-    for stmt, belief in zip(stmts, beliefs):
-        stmt.belief = belief
+
+    # Filter statements based on include_db_evidence
+    if not include_db_evidence:
+        stmts = [
+            stmt for stmt in stmts
+            if any(ev.source_api != 'database' for ev in stmt.evidence)
+        ]
+
     if deduplicate:
         # We do it this way to not change the order of the statements
         stmts = list({stmt.get_hash(): stmt for stmt in stmts}.values())
-    return stmts
-
-
-def generate_paper_clause(paper_term: Tuple[str, str]):
```

Comment:
Did this get moved somewhere else? The current tests error now because some imports of this function from here are failing. See for example https://github.com/gyorilab/indra_cogex/actions/runs/12262816912/job/34212968507?pr=188. 

You can run the tests that are currently failing locally to see if things work or not. These are the tests failing in the link above:

- test_analysis_integration.py
- test_enrichment_utils.py
- test_queries.py

Reply:
Added the generate_paper_clause function after the merge. 
Locally I am not facing any import errors and all the tests are passing. 

There is only one test which is failing i.e. def test_get_stmts_for_pmid 

Looking into why it's failing.

Reply:
Got all tests to work and pass. 

I found the issue with `get_stmts_for_paper`. In the new implementation, when the parameter logic was moved to `generate_paper_clause()`, we started using `$paper_parameter` in the query but were still passing it as `parameter=parameter` to `query_tx`. Since `query_tx` accepts parameters as keyword arguments, we need to match the parameter name exactly - so changing it to `paper_parameter=parameter` fixes the error